File: examples/snippets/modules/ROOT/examples/pekkostreams-java/src/main/java/cloudflow/pekkostreamsdoc/DataStreamingIngress.java
package cloudflow.pekkostreamsdoc;

// tag::httpStreamingIngress[]
import org.apache.pekko.http.javadsl.common.EntityStreamingSupport;
import org.apache.pekko.http.javadsl.marshallers.jackson.Jackson;

import cloudflow.pekkostream.PekkoServerStreamlet;

import cloudflow.pekkostream.PekkoStreamletLogic;
import cloudflow.pekkostream.util.javadsl.HttpServerLogic;
import cloudflow.streamlets.RoundRobinPartitioner;
import cloudflow.streamlets.StreamletShape;
import cloudflow.streamlets.avro.AvroOutlet;

public class DataStreamingIngress extends PekkoServerStreamlet {

  private AvroOutlet<Data> out =
      AvroOutlet.create("out", Data.class)
          .withPartitioner(RoundRobinPartitioner.getInstance());

  public StreamletShape shape() {
    return StreamletShape.createWithOutlets(out);
  }

  public PekkoStreamletLogic createLogic() {
    EntityStreamingSupport ess = EntityStreamingSupport.json();
    return HttpServerLogic.createDefaultStreaming(
        this, out, Jackson.byteStringUnmarshaller(Data.class), ess, getContext());
  }
}
// end::httpStreamingIngress[]

File: examples/snippets/modules/ROOT/examples/pekkostreams-scala/src/main/scala/cloudflow/pekkostreamsdoc/DataStreamingIngress.scala
package cloudflow.pekkostreamsdoc

// tag::httpStreamingIngress[]
import JsonSupport._
import org.apache.pekko.http.scaladsl.common.EntityStreamingSupport
import org.apache.pekko.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
import cloudflow.pekkostream._
import cloudflow.pekkostream.util.scaladsl._
import cloudflow.streamlets._
import cloudflow.streamlets.avro._

class DataStreamingIngress extends PekkoServerStreamlet {
  val out   = AvroOutlet[Data]("out", RoundRobinPartitioner)
  def shape = StreamletShape.withOutlets(out)

  implicit val entityStreamingSupport: EntityStreamingSupport = EntityStreamingSupport.json()
  override def createLogic                                    = HttpServerLogic.defaultStreaming(this, out)
}
// end::httpStreamingIngress[]
